SWENT-GRP5/SOLVIT
# Provider information Screen

### Description:
This pull request introduces the ProviderInfoScreen, which displays detailed information about individual service providers. This new UI screen offers users a comprehensive view of provider details, supporting informed decision-making when selecting service providers.

- **What issue does it solve?** This PR enhances the user experience by providing a dedicated screen to display essential provider information, allowing users to review provider details before engaging services.
- **What functionality does it add?** The ProviderInfoScreen presents provider details, including ratings, services offered, and other relevant information. MainActivity has been updated to navigate to this screen, and ListProviderViewModel and ServicesScreen now support selecting a provider and displaying their information.

### Related Issues:
- Fixes #262 

### Changes Made:
- Created `ProviderInfoScreen` to display detailed information for each provider.
- Updated `MainActivity` to support navigation to the new `ProviderInfoScreen`.
- Modified `ListProviderViewModel` to handle provider selection and `ServicesScreen` to trigger provider info display.
- Added unit tests for `ProviderInfoScreen` to ensure UI elements render correctly and data binding functions as expected.

### How to Test:
1. Select a provider from the list on the `ServicesScreen` and verify that the `ProviderInfoScreen` displays accurate provider details.
2. Run the tests in the `provider` package of the AndroidTeststo validate the functionality of `ProviderInfoScreen`.

---

### Checklist:
